### ISSUE TITLE
org-roam-buffer is empty in Doom emacs

### ISSUE THREAD

**Issue by dbjg** [jethrokuan/org-roam#179]
#### Steps to Reproduce

1. Load Emacs
2. Run `org-roam`

#### Expected Results

The org-roam-buffer should show the backlinks.
#### Actual Results

The backlinks window is empty.
![Screenshot_2020-02-24_20-00-32](https://user-images.githubusercontent.com/61430429/75182369-89d23680-5740-11ea-84f4-6860c0911465.png)

### Versions
- Emacs: v26.3
- Org-roam commit: https://github.com/jethrokuan/org-roam/commit/d780b6ffd176fbbfe863c2fccfba0f29e1b60d0d

**Comment by dbjg** [comment]
If I can get this to work I'll make a PR to add documentation on how to install org-roam in Doom emacs (just like there is for Spacemacs).

**Comment by jethrokuan** [comment]
This has been repeated a number of times, but you probably didn't turn on org-roam-mode. I'm going to make the `org-roam` toggle the minor mode if it's not on.

**Comment by gperetin** [comment]
I had the same problem, with Doom as well, the issue in my case was that I was using the master branch, which at the time (this was 3-4 days ago) was not working for some reason. Updating to develop fixed this for me.

Here's how I have it set up, and it's working:
- in packages.el
```
(package! org-roam
  :recipe (:host github :repo "jethrokuan/org-roam" :branch "develop"))
```

- in config.el
```
(use-package! org-roam

  :hook ((org-mode . org-roam-mode)
         (after-init . org-roam--build-cache-async))
  :config
  (setq org-roam-directory "~/Notes/roam")

  (map!
   "C-c n l" #'org-roam
   "C-c n t" #'org-roam-today
   "C-c n f" #'org-roam-find-file
   "C-c n i" #'org-roam-insert
   "C-c n g" #'org-roam-show-graph
   )
  )
```

**Comment by yangnom** [comment]
> I had the same problem, with Doom as well, the issue in my case was that I was using the master branch, which at the time (this was 3-4 days ago) was not working for some reason. Updating to develop fixed this for me.
> 
> Here's how I have it set up, and it's working:
> 
> * in packages.el
> 
> ```
> (package! org-roam
>   :recipe (:host github :repo "jethrokuan/org-roam" :branch "develop"))
> ```
> 
> * in config.el
> 
> ```
> (use-package! org-roam
> 
>   :hook ((org-mode . org-roam-mode)
>          (after-init . org-roam--build-cache-async))
>   :config
>   (setq org-roam-directory "~/Notes/roam")
> 
>   (map!
>    "C-c n l" #'org-roam
>    "C-c n t" #'org-roam-today
>    "C-c n f" #'org-roam-find-file
>    "C-c n i" #'org-roam-insert
>    "C-c n g" #'org-roam-show-graph
>    )
>   )
> ```

This might be worth throwing into the documentation.  I'm probably biased because I use DOOM as well, but it's nice to have this kind of info up, and note that the `(setq org-roam-directory "~/Notes/roam")` needs to point to the directory in which you wish to keep all your roam notes.

**Comment by bdarcus** [comment]
> :hook ((org-mode . org-roam-mode)
>          (after-init . org-roam--build-cache-async))

The generic and spacemacs instructions have the line `(after-init . org-roam-mode)` for the hook. What's the significance of the difference?

**Comment by jethrokuan** [comment]
`org-roam-mode` now needs to also advice `rename-file` and `delete-file`, and we want this advice to happen only once. Therefore `org-roam-mode` needed to be come a global minor-mode to control the addition and removal of the advice that keeps the cache consistent. This change happened in `0.1.2`, which master is also on now.

**Comment by dbjg** [comment]
I have the above settings from @yangnom in the correct files (I had almost the same but just to be sure). I even ran the cache manually. And the org-roam-mode is enabled. 

org-roam-insert and org-roam-find-file work properly but the org-roam window on the right is still empty.

**Comment by jethrokuan** [comment]
The settings above is now incorrect, as reflected in the updated installation instructions:

```
(use-package! org-roam
  :hook (after-init . org-roam-mode)
...

is now the way to install it.

**Comment by dbjg** [comment]
Ah I see. I'm sorry for not reading properly, my mistake.

Within now and a few days I'll make a PR for the added documentation. Thanks so much for building this package!

**Comment by bdarcus** [comment]
I don't know if this is exactly right, but this is what I currently have. I defined the keybindings in the bindings.el file.

```elisp
(use-package! org-roam
  :after org
  :hook (after-init . org-roam-mode)
  :config
  (setq org-roam-directory "~/org/roam"))
```

**Comment by technovangelist** [comment]
so i have no idea what i am doing since org-roam is my intro to org mode and my introduction to emacs in general. 

in config.el: 
`(use-package org-roam
  :hook
  (after-init . org-roam-mode)
  :custom
  (org-roam-directory "~/org-roam")
  :bind (:map org-roam-mode-map
          (("<leader> n r" . org-roam)
           ("C-c n f" . org-roam-find-file)
           ("C-c n g" . org-roam-show-graph))
          :map org-mode-map
          (("C-c n i" . org-roam-insert))
          )
  )

;; this part lets me press space r to get roam related stuff.
(map! :leader
      (:prefix ("r" . "org mode")
        :desc "Open Org Mode" "o" #'org-mode
        :desc "Open Org Roam" "r" #'org-roam
        :desc "Insert to Roam" "i" #'org-roam-insert
        :desc "find Roam File" "f" #'org-roam-find-file
        :desc "deft" "d" #'deft
        :desc "New Journal Entry" "j" #'org-journal-new-entry
      ))

(use-package deft
  :after org
  :custom
  (deft-recursive t)
  (deft-use-filter-string-for-filename t)
  (deft-default-extension "org")
  (deft-directory "~/org-roam"))

(use-package org-journal
  :custom
  (org-journal-date-prefix "#+TITLE: ")
  (org-journal-file-format "%Y-%m-%d.org")
  (org-journal-dir "~/org-roam")
  (org-journal-date-format "%A, %d %B %Y"))
`
and in packages.el: 
`(package! org-roam :recipe (:host github :repo "jethrokuan/org-roam" :branch "develop"))
`

**Comment by bdarcus** [comment]
Since you have those keybindings specified outside use-package!, you should be able to remove them above; as I have (I put the keybindings in a separate bindings.el file).

Again: am not an expert.
